app/features/quizzify/tools.py

```diff
@@ -102,61 +358,111 @@ def load(self) -> List[Document]:
 
 class URLLoader:
-    def __init__(self, file_loader=None, expected_file_type="pdf", verbose=False):
-        self.loader = file_loader or BytesFilePDFLoader
-        self.expected_file_type = expected_file_type
+    def __init__(self, file_loader=None, expected_file_types=None, verbose=False):
+        self.loader = file_loader or BytesFileLoader
+        self.expected_file_types = expected_file_types or ["pdf", "doc", "docx", "ppt", "pptx", "csv"]
         self.verbose = verbose
-    
-    def load(self, tool_files: list[ToolFile]) -> list[Document]:
+
+    def load(self, tool_files: List[ToolFile]) -> List[Document]:
         queued_files = []
-        
+        documents = []
+        any_success = False
+
         for tool_file in tool_files:
-            url = tool_file.url
-            response = requests.get(url)
-            
-            parsed_url = urlparse(url)
-            path = parsed_url.path
-            
-            if response.status_code == 200:
-                # Read file
-                file_content = BytesIO(response.content)
-                
-                # Check file type
-                file_type = path.split(".")[-1]
-                if file_type != self.expected_file_type: 
-                    raise ValueError(f"Expected file type: {self.expected_file_type}, but got: {file_type}")
-                
-                # Append to Queue
-                queued_files.append((file_content, file_type))
-                if self.verbose: logger.info(f"Successfully loaded file from {url}")
-                
-            else: logger.error(f"Request Failed to load file from {url}")
-            
-            # Pass Queue to the file loader
+            try:
+                url = tool_file.url
+                response = requests.get(url)
+                parsed_url = urlparse(url)                
+                path = parsed_url.path
+                                
+                if path.endswith(('.pdf', '.doc', '.docx', '.ppt', '.pptx', '.csv')):
+                                      
+                    if response.status_code == 200:
+                        file_content = BytesIO(response.content)
+                        file_type = tool_file.filename.split(".")[-1].lower()
+                        if file_type not in self.expected_file_types:
+                            raise LoaderError(f"Expected file types: {self.expected_file_types}, but got: {file_type}")
+                        queued_files.append((file_content, file_type))
+                        if self.verbose:
+                            logger.info(f"Successfully loaded file from {url}")
+                        any_success = True
+                    else:
+                        logger.error(f"Request failed to load file from {url} and got status code {response.status_code}")
+
+                elif parsed_url.netloc in ["youtu.be","m.youtube.com","youtube.com","www.youtube.com","www.youtube-nocookie.com","vid.plus"]:
```

Comment:
I think you defined a set of valid URL slugs earlier? We can use that list here

Reply:
Yes! We missed it and its been resolved.